src/corelibs/U2Gui/src/Notification.cpp

```diff
@@ -453,4 +356,68 @@ bool NotificationStack::eventFilter(QObject* target, QEvent* event) {
     return false;
 }
 
+void NotificationStack::activate(Notification* notification) {
+    SAFE_POINT(notification, "Notification is null", );
+    QString text = notification->text;
+    QAction* action = notification->action;
+    if (floatingNotifications.contains(notification)) {
+        remove(notification);
+    }
+    if (action) {
+        action->trigger();
+        return;
+    }
+
+    QObjectScopedPointer<QDialog> dlg = new QDialog(AppContext::getMainWindow()->getQMainWindow());
+    dlg->setObjectName("NotificationDialog");
+    QVBoxLayout vLayout;
+    QHBoxLayout hLayout;
+    QPushButton ok;
+    QCheckBox isDelete;
+
+    ok.setText(tr("OK"));
+    isDelete.setText(tr("Remove notification after closing"));
+    isDelete.setChecked(true);
+    connect(&ok, SIGNAL(clicked()), dlg.data(), SLOT(accept()));
+    hLayout.addWidget(&isDelete);
+    hLayout.addWidget(&ok);
+
+    dlg->setLayout(&vLayout);
+    QTextBrowser txtEdit;
+    txtEdit.setOpenExternalLinks(true);
+    txtEdit.setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+    dlg->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+    txtEdit.setReadOnly(true);
+    txtEdit.setText(text);
+    vLayout.addWidget(&txtEdit);
+    vLayout.addLayout(&hLayout);
+
+    dlg->setWindowTitle(tr("Detailed message"));
+
+    setFixed(true);
+    int dialogResult = dlg->exec();
+    setFixed(false);
+    CHECK(!dlg.isNull(), );
+
+    if (dialogResult == QDialog::Accepted && isDelete.isChecked() && notifications.contains(notification)) {
+        remove(notification);
+    }
+}
+
+void NotificationStack::sl_updateNotificationState() {
+    QList<Notification*> notificationsToRemove;
+    for (auto notification : qAsConst(floatingNotifications)) {
+        qint64 onScreenTimeMillis = QDateTime::currentMSecsSinceEpoch() - notification->timestamp;
+        if (onScreenTimeMillis >= Notification::ON_SCREEN_TIMEOUT_MILLIS) {
+            notificationsToRemove.append(notification);
+        }
+    }
+    CHECK(!notificationsToRemove.isEmpty(), );
+    for (auto notification : qAsConst(notificationsToRemove)) {
+        floatingNotifications.removeOne(notification);
```

Comment:
Shouldn't you use `remove(Notification*)` here?

Reply:
I got, that you call `updateOnScreenNotificationPositions()` several times in this case, but, I suppose, it's better to use the same deletion way [at this line](https://github.com/ugeneunipro/ugene/pull/875/files#diff-0ac7ab4346eced218a224dbce7a4950cc03261784fd3d3b32d2732345d42b82dR274) as here too.

Reply:
I prefer atomic delete here with no in-the-middle re-layout of the visible UI stack on every delete.

The other line you mention is used to delete embedded notifications that have a trickier code and in that case almost never trigger re-layout or the floating stack - so there is a reason for a code-reuse.